va/va_trace.h

```diff
@@ -54,6 +57,127 @@ extern int va_trace_flag;
         va_TraceStatus(dpy, __func__, ret);     \
     }
 
+/* event id definition */
+enum {
+    INVALIDE_ID = 0,
+    CREATE_CONFIG = 1,
+    DESTROY_CONFIG,
+    CREATE_CONTEXT,
+    DESTROY_CONTEXT,
+    CREATE_BUFFER,
+    DESTROY_BUFFER,
+    CREATE_SURFACE,
+    DESTROY_SURFACE,
+    BEGIN_PICTURE,
+    RENDER_PICTURE,
+    END_PICTURE,
+    BUFFER_DATA,
+    SYNC_SURFACE,
+    SYNC_SURFACE2,
+    QUERY_SURFACE_ATTR,
+};
+
+/* event opcode definition */
+#define TRACE_INFO  0
+#define TRACE_BEGIN 1
+#define TRACE_END   2
+#define TRACE_DATA  3
+
+typedef struct va_event_data {
+    void *buf;
+    unsigned int size;
+} VAEventData;
+
+#define VA_TRACE(dpy,id,op)                         \
```

Comment:
wrap trace interface to minimal runtime perf impact, only process data when enabled.